### PR TITLE
feat: Allow choosing between a basic and an advanced plugin template.

### PR DESCRIPTION
**Pull request by misteroneill**

### CODE REVIEW

File: generators/app/templates/src/_plugin-advanced.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+   *         An optional options object.
+   *
+   *         While not a core part of the Video.js plugin architecture, a
+   *         second argument of options is a convenient way to accept inputs
+   *         from your plugin's caller.
+   */
+  constructor(player, options) {
+    super(player);
+
+    this.options = videojs.mergeOptions(defaults, options);
```
Comment (BrandonOCasey) [inline generators/app/templates/src/_plugin-advanced.js]: Maybe add a `this.player = player` here too?

Comment (misteroneill) [reply generators/app/templates/src/_plugin-advanced.js]: That happens implicitly in `super()`. Maybe I'll add a comment indicating that `this.player` is available.

File: generators/app/constants.js [hunk 2]
```diff
@@ -48,6 +53,7 @@ module.exports = {
     husky: 'lint',
     lang: false,
     license: LICENSE_DEFAULT,
+    pluginType: 'basic',
```
Comment (gkatsev) [inline generators/app/constants.js]: I think the default should be "advanced".

File: generators/app/templates/src/_plugin-advanced.js (hunk 1 above)
Comment (gkatsev) [reply generators/app/templates/src/_plugin-advanced.js]: We should link to the docs and blog posts and such.

**Pull request merged**